### ISSUE TITLE
Generated migrations drop default after setting

### ISSUE THREAD

**Issue by Playfrog4u** [typeorm/typeorm#9950]
### Issue description

Generated migrations drop default after setting

### Expected Behavior

When generating a migration, it seems like it should keep the PrimaryGeneratedColumn default and not drop it.

### Actual Behavior

When generating a migration file after I rename a table in postgres and rename the table within the entity, it seems to set the default but then immediately drop the default as seen in the migration below.. 

```
import { MigrationInterface, QueryRunner } from "typeorm";

export class PostRefactoring1681516030326 implements MigrationInterface {
    name = 'PostRefactoring1681516030326'

    public async up(queryRunner: QueryRunner): Promise<void> {
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_18ec13e3aea890ecdbfa8a170fd"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_3aea4fdf5c3821f188e3ec63540"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_78c23de3caac7025ce84d546ac2"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_a4a7f8a7ad0258e936d1ba46729"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_acaad7a4e7d46432d53a949deba"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_e187b1dbea1c8f988a710815bb7"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_e435a25a2dc031a21d32524b0d7"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_e9d034432e13031cdc6955f6672"`);
        await queryRunner.query(`CREATE SEQUENCE IF NOT EXISTS "table_old_id_seq" OWNED BY "table_old"."id"`);
        await queryRunner.query(`ALTER TABLE "table_old" ALTER COLUMN "id" SET DEFAULT nextval('"table_old_id_seq"')`);
        await queryRunner.query(`ALTER TABLE "table_old" ALTER COLUMN "id" DROP DEFAULT`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_63fa3eb0bf2e8e4a058a1f009bf" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_81f7385ca8a1fa90559d67f6315" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_5cdf6d059e63ca0dce5d85e80f4" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_d9febe9cadb31be5fbbf74b9b4e" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_aee6af933dc00312e643aad5fe8" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_32b6d490326cbc85891fbbe324c" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_b589b44d296ce541e1891d527a8" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_b0e92e4000a0463256bdca307eb" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
    }

    public async down(queryRunner: QueryRunner): Promise<void> {
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_b0e92e4000a0463256bdca307eb"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_b589b44d296ce541e1891d527a8"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_32b6d490326cbc85891fbbe324c"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_aee6af933dc00312e643aad5fe8"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_d9febe9cadb31be5fbbf74b9b4e"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_5cdf6d059e63ca0dce5d85e80f4"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_81f7385ca8a1fa90559d67f6315"`);
        await queryRunner.query(`ALTER TABLE "table_old" DROP CONSTRAINT "FK_63fa3eb0bf2e8e4a058a1f009bf"`);
        await queryRunner.query(`ALTER TABLE "table_old" ALTER COLUMN "id" SET DEFAULT nextval('xxx_id_seq')`);
        await queryRunner.query(`ALTER TABLE "table_old" ALTER COLUMN "id" DROP DEFAULT`);
        await queryRunner.query(`DROP SEQUENCE "table_old_id_seq"`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_e9d034432e13031cdc6955f6672" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_e435a25a2dc031a21d32524b0d7" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_e187b1dbea1c8f988a710815bb7" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_acaad7a4e7d46432d53a949deba" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_a4a7f8a7ad0258e936d1ba46729" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_78c23de3caac7025ce84d546ac2" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_3aea4fdf5c3821f188e3ec63540" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
        await queryRunner.query(`ALTER TABLE "table_old" ADD CONSTRAINT "FK_18ec13e3aea890ecdbfa8a170fd" FOREIGN KEY ("xxx_id") REFERENCES "yyy"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
    }

}
```

### Steps to reproduce

When rename a table directly in postgres then update the entity name to match the change, then run a generate migration it does not keep the default on the PrimaryGeneratedColumn

### My Environment

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |  MacOS 13.3        |
| Node.js version     | v16.19.1 |
| Typescript version  |^0.3.14  |
| TypeORM version     | ~4.8.2  |

### Relevant Database Driver(s)

- [ ] aurora-mysql
- [ ] aurora-postgres
- [ ] better-sqlite3
- [ ] cockroachdb
- [ ] cordova
- [ ] expo
- [ ] mongodb
- [ ] mysql
- [ ] nativescript
- [ ] oracle
- [X] postgres
- [ ] react-native
- [ ] sap
- [ ] spanner
- [ ] sqlite
- [ ] sqlite-abstract
- [ ] sqljs
- [ ] sqlserver

### Are you willing to resolve this issue by submitting a Pull Request?

No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by patrykpieniak** [comment]
I have the same problem on the latest version (0.3.15). TypeORM generates migrations and synchronization badly.

In my case, I have a MariaDB database.

I discovered that the problem is the function: compareDefaultValues
I have a relation, in the database the default value is "null", while TypeORM says "undefined". This is what the logs show.

![image](https://user-images.githubusercontent.com/1338690/232238538-7c8ee422-b78d-40a8-9958-2a75db2382a1.png)
![image](https://user-images.githubusercontent.com/1338690/232238550-8f06569b-7a8b-4e13-957c-f4d2e116e591.png)

By contrast, migration correctly generates and uses "null"
![image](https://user-images.githubusercontent.com/1338690/232238588-927d30ff-cb45-4314-b197-0681ec417f4a.png)


I don't know why it hasn't been fixed. This has been a known problem for a long time.

**Comment by N-Andronopoulos** [comment]
I believe it's related to #10090 in 0.3.16 it happens (for me) in synchronize phase too.

**Comment by pleerock** [comment]
Must be fixed by #10041
